### ISSUE TITLE
Configuration file problems

### ISSUE THREAD

**Issue by jonas054** [rubocop/rubocop#116]
I think we need to handle configuration files in a better way. The problems with the current implementation are:
- We're too soft on missing configuration. Every cop parameter, including `Enabled`, has a default value. This means that when users update the rubocop gem and get a number of new cops, their old `.rubocop.yml` files in the projects should be updated too, but they might forget and they won't see the need for filling in more configuration values because of the defaults.
- There's no error or warning for unexpected configuration values. A misspelled or outdated cop name or configuration parameter will go unnoticed.
- If you have many `.rubocop.yml` files, you have a lot of duplication.

I have an idea for a solution. We add a `config` directory to the rubocop root. Here we store default configuration files such as `all_enabled.yml` which will look like the `.rubocop.yml` in the root today. Then we add an include mechanism called `based_on`. E.g.:

``` yaml
# .rubocop.yml

based_on: RUBOCOP_HOME/config/all_enabled.yml

LineLength:
  Enabled: true
  Max: 100
```

So everything after `based_on` will override settings in the included yml file. If you override the settings for a cop, you must specify all parameters, i.e. you can't just set `Max` and expect to inherit `Enabled`.

And we remove the default handling. Everything must be specified.
